nussl/core/augmentation.py

```diff
@@ -0,0 +1,377 @@
+import numpy as np
+import numpy.random as random
+import librosa
+import os
+from .audio_signal import AudioSignal
+import tempfile
+from .augmentation_utils import *
+
+# These values are found in the ffmpeg documentation in filters
+# that use the level_in arugment, however, not all filters that
+# use this state these bounds in documentationwith pytest.raises(ValueError):
+LEVEL_MIN = .015625
+LEVEL_MAX = 64
+
+def time_stretch(audio_signal, stretch_factor):
+    """
+    Linear Stretch on the time axis
+    Args: 
+        audio_signal: An AudioSignal object
+        factor_range: A tuple of length 2. Denotes start and end of possible ranges for factor. 
+    Returns:
+        stretched_signal: A copy of the original audio signal, with augmentations applied 
+    """
+    if not np.issubdtype(type(stretch_factor), np.number) or stretch_factor <= 0:
+        raise ValueError("stretch_factor must be a positve scalar")
+
+    sample_rate = audio_signal.sample_rate
+    stretched_audio_data = []
+    audio_data = audio_signal.audio_data
+
+    for audio_row in audio_signal.get_channels():
+        stretched_audio_data.append(librosa.effects.time_stretch(audio_row, stretch_factor))
+    stretched_audio_data = np.array(stretched_audio_data)
+
+    # The following line causes a UserWarning.
+    # stretched_signal = audio_signal.make_copy_with_audio_data(stretched_audio_data)
+
+    # This one does not
+    stretched_signal = AudioSignal(audio_data_array=stretched_audio_data)
+    stretched_signal.stft()
```

Comment:
Why do we need to take the STFT here?

Reply:
If you make a copy, do you not have to take the STFT? I thought that did not automatically populate?

Reply:
The user should take the STFT instead, otherwise we are doing some side computation that the user didn't request and may not need. Function should only do what is specified.